src/sentry/utils/performance_issues/detectors/query_injection_detector.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+import hashlib
+from typing import Any
+
+from sentry.issues.grouptype import DBQueryInjectionVulnerabilityGroupType
+from sentry.issues.issue_occurrence import IssueEvidence
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.utils import json
+from sentry.utils.performance_issues.base import (
+    DetectorType,
+    PerformanceDetector,
+    get_notification_attachment_body,
+)
+from sentry.utils.performance_issues.performance_problem import PerformanceProblem
+from sentry.utils.performance_issues.types import Span
+
+MAX_EVIDENCE_VALUE_LENGTH = 10_000
+
+
+class QueryInjectionDetector(PerformanceDetector):
+    __slots__ = "stored_problems"
+
+    type = DetectorType.QUERY_INJECTION
+    settings_key = DetectorType.QUERY_INJECTION
+
+    def __init__(self, settings: dict[DetectorType, Any], event: dict[str, Any]) -> None:
+        super().__init__(settings, event)
+
+        self.stored_problems = {}
+        self.potential_unsafe_inputs: list[tuple[str, dict[str, Any]]] = []
+        self.extract_request_data(event)
+
+    def extract_request_data(self, event: dict[str, Any]) -> None:
+        self.request_data = event.get("request", {}).get("data", {})
+        self.request_url = event.get("request", {}).get("url", "")
+        if not isinstance(self.request_data, dict):
+            return
+
+        for query_pair in self.request_data.items():
```

Comment:
```suggestion
        for key, value in self.request_data.items():
```
or something like that.

Reply:
i think i prefer to not to unpack here, since the key isn't used and we want to keep both values .